test/FeatureFlagTest.php

```diff
@@ -0,0 +1,2796 @@
+        ];
+
+        self::assertTrue(FeatureFlag::matchProperty($prop, [
+            "key" => 1,
+        ]));
+
+        self::assertTrue(FeatureFlag::matchProperty($prop, [
+            "key" => 2,
+        ]));
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => 0,
+        ]));
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => -1,
+        ]));
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => "3",
+        ]));
+
+        $prop = [
+            "key" => "key",
+            "value" => 43,
+            "operator" => "lte"
+        ];
+
+        self::assertTrue(FeatureFlag::matchProperty($prop, [
+            "key" => 0,
+        ]));
+
+        self::assertTrue(FeatureFlag::matchProperty($prop, [
+            "key" => 43,
+        ]));
+
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => 44,
+        ]));
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => "1",
+        ]));
+
+        self::assertFalse(FeatureFlag::matchProperty($prop, [
+            "key" => "3",
+        ]));
+    }
+
+    public function testFlagPersonProperties()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse: MockedResponses::LOCAL_EVALUATION_REQUEST);
+        $this->client = new Client(
+            FAKE_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client,
+            "test"
+        );
+        PostHog::init(null, null, $this->client);
+
+        $this->assertTrue(PostHog::getFeatureFlag('person-flag', 'some-distinct-id', false, [], ["region" => "USA"]));
+        $this->assertFalse(PostHog::getFeatureFlag('person-flag', 'some-distinct-id-2', false, [], ["region" => "Canada"]));
+    }
+
+    public function testFlagGroupProperties()
+    {
+        $this->http_client = new MockedHttpClient(host: "app.posthog.com", flagEndpointResponse: MockedResponses::LOCAL_EVALUATION_GROUP_PROPERTIES_REQUEST);
+        $this->client = new Client(
+            FAKE_API_KEY,
+            [
+                "debug" => true,
+            ],
+            $this->http_client,
+            "test"
+        );
+        PostHog::init(null, null, $this->client);
+
+        $this->assertFalse(PostHog::getFeatureFlag('group-flag', 'some-distinct-1', false, [], [], ["company" => ["name" => "Project Name 1"]]));
+        $this->assertFalse(PostHog::getFeatureFlag('group-flag', 'some-distinct-2', false, [], [], ["company" => ["name" => "Project Name 2"]]));
+        $this->assertTrue(PostHog::getFeatureFlag('group-flag', 'some-distinct-id', false, ["company" => "amazon_without_rollout"], [], ["company" => ["name" => "Project Name 1"]]));
+        $this->assertFalse(PostHog::getFeatureFlag('group-flag', 'some-distinct-i', false, ["company" => "amazon"], [], ["company" => ["name" => "Project Name 1"]]));
```

Comment:
I assume this test was passing earlier (without the `$groups[$groupsName]` change because of this?

```suggestion
        $this->assertFalse(PostHog::getFeatureFlag('group-flag', 'some-distinct-id', false, ["company" => "amazon"], [], ["company" => ["name" => "Project Name 1"]]));
```